ClemBot.Bot/bot/cogs/trivia_cog.py

```diff
@@ -0,0 +1,566 @@
+
+        await self.bot.messenger.publish(Events.on_set_pageable_embed,
+                                         pages=final_page,
+                                         author=ctx.author,
+                                         channel=ctx.channel,
+                                         timeout=60, )
+
+        return
+
+    async def url_builder(self, function_list, input_length):
+
+        max_index = input_length - 1
+        url = URL_BUILDER + str(function_list[0])
+        x = 1
+
+
+        while x <= max_index:
+            if function_list[x]:
+                match x:
+                    case 1:
+                        url += ("&category=" + str(function_list[1]))
+                    case 2:
+                        url += ("&difficulty=" + function_list[2])
+                    case 3:
+                        url += ("&type=" + function_list[max_index])
+            x += 1
+
+        return url
+
+    async def matching_function(self, case, *input: str):
+
+        match case:  #Revolves around beautiful O(1) based indexing
+            case 0:
+                if input[0].isnumeric():
+                    question_number = int(input[0])
+
+                    if 0 < question_number <= 50:
+                        return question_number
+                else:
+                    raise UserInputError(
+                        "Question Number has to be a number within the range of 1 to 50")
+            case 1:
+                if input[1].isnumeric():
+                    trivia_number = int(input[1])
+                    if 0 < trivia_number <= 24:
+                        return trivia_number + 8
+                    elif trivia_number == 0:
+                        return None
+                    else:
+                        raise UserInputError(
+                            "Category Number out of bounds(Number has to be 1-24) or enter the category you want! Type ?trivia help to see the category list")
+                else:
+                    trivia_category = input[1].lower()
+
+                    for x in CATEGORYLIST_LOWER:
+                        if x.find(trivia_category) != -1:
+                            return_this = CATEGORYLIST_LOWER.index(x) + 9
+                            return return_this
+                    else:
+                        raise UserInputError("Category not found!")
+            case 2:
+                if input[2].isnumeric():
+                    evaluate_int = int(input[2])
+                    if 0 < evaluate_int <= 3:
+                        return_string = DIFFICULTY_LOWER[evaluate_int - 1]
+                        return return_string
+                    elif evaluate_int == 0:
+                        return None
+                    else:
+                        raise UserInputError(
+                            "Difficulty Number out of bounds(Number has to be 1-3) or enter Easy-Hard! Type ?trivia help to see the difficulty list.")
```

Comment:
We need to grab the actual command prefix for the context guild and interpolate it into this string